File: February/Java/Divide Array Into Arrays With Max Difference./Divide Array Into Arrays With Max Difference.java
class Solution {
        public int[][] divideArray(int[] nums, int k) {
                int len = nums.length;
        int[][] res = new int[len/3][3];
        Arrays.sort(nums);
        for(int i=0,m=0; i<len; i=i+3,m++){
            res[m][0] = nums[i];
            for(int j=1; j<3; j++){
                if(nums[i+j]-res[m][0]<=k){
                    res[m][j]=nums[i+j];
                } else{
                    return new int[0][0];
                }
            }
        }
        return res;
    }
}

File: February/Python/Divide Array Into Arrays With Max Difference/Divide Array Into Arrays With Max Difference.py
class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        nums.sort()
        ans = []
        for i in range(0, len(nums), 3):
            if nums[i + 2] - nums[i] > k:
                return []
            ans.append([nums[i], nums[i + 1], nums[i + 2]])
        return ans
